### PR TITLE
Embed video in modal

### PR DESCRIPTION
**Pull request by gebederry**

### CODE REVIEW

File: assets/js/scripts.js [hunk 1]
```diff
@@ -1043,6 +1043,30 @@ function formatLootTableLevel(table, rate = 1, level = 0) {
...
+  const url = button.getAttribute('data-video-url');
+  if (url.includes('youtube')) {
+    const videoId = url.match(/v=([^&]+)/)[1];
+    const startTime = url.match(/t=(\d+)/)?.[1] || 0;
+    videoFrameEl.src = `https://www.youtube.com/embed/${videoId}?autoplay=1&start=${startTime}`;
+  } else if (url.includes('bilibili')) {
+    const videoId = url.match(/\/video\/([^\/?]+)/)[1];
+    const startTime = url.match(/t=(\d+)/)?.[1] || 0;
+    videoFrameEl.src = `https://player.bilibili.com/player.html?bvid=${videoId}&high_quality=1&t=${startTime}`;
+  }
```
Comment (MichalD96) [inline assets/js/scripts.js:1064]: Just a suggestion. Wouldn't be better to use URL object?

```javascript
  const url = new URL(button.getAttribute('data-video-url')); // Assumming attribute is not null or empty string

  if (['youtube', 'youtu.be'].some(value => url.host.includes(value))) {
    const videoId = url.searchParams.get('v');
    const startTime = url.searchParams.get('t') || 0;
    videoFrameEl.src = `https://www.youtube.com/embed/${videoId}?autoplay=1&start=${startTime}`;
    console.log(videoFrameEl.src);
  } else if (url.host.includes('bilibili')) {
    const [, videoId] = url.pathname.match(/\/video\/([^\/?]+)/); //? /\/video\/(\w+)/
    const startTime = url.searchParams.get('t') || 0;
    videoFrameEl.src = `https://player.bilibili.com/player.html?bvid=${videoId}&high_quality=1&t=${startTime}`;
  }
```

File: assets/css/styles.css
Comment (MichalD96) [inline assets/css/styles.css:1]: It looks good on PC, but needs to be 100% width on mobile devices.

File: assets/js/scripts.js (hunk 1 above)
Comment (gebederry) [reply assets/js/scripts.js:1064]: Thanks for correction, more modern approach.

File: assets/css/styles.css [hunk 2]
```diff
@@ -1326,6 +1334,20 @@ a.no-style:visited {
     }
 }
 
+#video-modal .modal-xl {
+    width: 66.667%;
+}
+
+@media (max-width: 768px) {
+    #video-modal .modal-xl {
+        width: 100%;
```
Comment (MichalD96) [inline assets/css/styles.css:1343]: Just a detail. For some reason dialog container overflows on mobile devices.
Other modals doesn't have that problem. The easiest fix would be:

```css
width: calc(100vw - 15px);
```

but the reason of that overflow is somewhere else.

**Review by MichalD96 — APPROVED** [review]

**Pull request merged**